File: src/main/java/me/tsukanov/counter/domain/CounterApplicationComponent.java
package me.tsukanov.counter.domain;

import dagger.Component;
import dagger.android.AndroidInjectionModule;
import dagger.android.AndroidInjector;
import javax.inject.Singleton;
import me.tsukanov.counter.CounterApplication;
import me.tsukanov.counter.repository.CounterStorage;

@Singleton
@Component(modules = { AndroidInjectionModule.class, CounterModule.class})
public interface CounterApplicationComponent extends AndroidInjector<CounterApplication> {

  CounterStorage localStorage();
}

File: app/src/main/java/me/tsukanov/counter/domain/CounterApplicationComponent.kt
package me.tsukanov.counter.domain

import dagger.Component
import dagger.android.AndroidInjectionModule
import dagger.android.AndroidInjector
import me.tsukanov.counter.CounterApplication
import me.tsukanov.counter.repository.CounterStorage
import javax.inject.Singleton

@Singleton
@Component(
    modules = [
        AndroidInjectionModule::class,
        CounterModule::class,
    ]
)
interface CounterApplicationComponent : AndroidInjector<CounterApplication?> {

    fun localStorage(): CounterStorage<IntegerCounter?>?

}
